java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/streamreactor/connect/azure/eventhubs/util/KcqlConfigPortTest.java

```diff
@@ -1,23 +1,73 @@
+import static org.junit.jupiter.api.Assertions.fail;
 
-import io.lenses.kcql.Kcql;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.config.ConfigException;
 import org.junit.jupiter.api.Test;
 
 class KcqlConfigPortTest {
 
   @Test
-  void parseMultipleKcqlStatementsPickingOnlyFirstShouldDropSecondStatementAndParseFirst() {
+  void mapInputToOutputsFromConfigForMultipleKcqlStatementsShouldRetunMapOfInputToOutput() {
     //given
-    String kcql1 = "insert into output1 select * from input1";
-    String kcql2 = "insert into output2 select * from input2";
-    String bothStatements = kcql1 + "; " + kcql2 + ";";
+    int numberOfMappings = 3;
+    List<String> inputs = new ArrayList<>(numberOfMappings);
+    List<String> outputs = new ArrayList<>(numberOfMappings);
+    String kcqlTemplate = "insert into %s select * from %s;";
+    StringBuilder fullKcql = new StringBuilder();
 
+    for (int i = 0; i < numberOfMappings; i++) {
+      String newInput = "INPUT" + i;
+      String newOutput = "OUTPUT" + i;
+
+      inputs.add(i, newInput);
+      outputs.add(i, newOutput);
+      fullKcql.append(String.format(kcqlTemplate, newOutput, newInput));
+    }
     //when
-    Kcql kcql = KcqlConfigPort.parseMultipleKcqlStatementsPickingOnlyFirst(bothStatements);
+    Map<String, String> inputToOutputsFromConfig = KcqlConfigPort.mapInputToOutputsFromConfig(
+        fullKcql.toString());
 
     //then
-    assertEquals(kcql1, kcql.getQuery());
+    for (String input : inputToOutputsFromConfig.keySet()){
+      int indexOfInput = inputs.indexOf(input);
+      assertNotEquals(-1, indexOfInput);
+      assertEquals(inputs.get(indexOfInput), input);
+      assertEquals(outputs.get(indexOfInput), inputToOutputsFromConfig.get(input));
+    }
+  }
+
+  @Test
+  void mapInputToOutputsFromConfigShouldntAllowForIllegalNames() {
+    //given
+    String illegalInputKcql = "INSERT INTO OUTPUT SELECT * FROM 'INPUT*_'";
+    String illegalOutputKcql = "INSERT INTO 'OUTPUT*_' SELECT * FROM INPUT";
+    String inputErrorMessage = "Input topic INPUT*_, name is not correctly specified "
+        + "(It can contain only letters, numbers and hyphens, underscores and "
+        + "dots and has to start with number or letter";
+    String outputErrorMessage = "Output topic OUTPUT*_, name is not correctly specified "
+        + "(It can contain only letters, numbers and hyphens, underscores and "
+        + "dots and has to start with number or letter";
+
+    //when
+    try {
+      KcqlConfigPort.mapInputToOutputsFromConfig(illegalInputKcql);
+      fail("Exception not thrown");
+    } catch (ConfigException e) {
+      assertEquals(inputErrorMessage, e.getMessage());
+    }
+
+    try {
+      KcqlConfigPort.mapInputToOutputsFromConfig(illegalOutputKcql);
+      fail("Exception not thrown");
+    } catch (ConfigException e) {
+      assertEquals(outputErrorMessage, e.getMessage());
+    }
+
```

Comment:
I think there needs to be a few more tests:

```
INSERT INTO a SELECT * FROM x;
INSERT INTO a SELECT * FROM y;
```
should be accepted


```
INSERT INTO a SELECT * FROM x;
INSERT INTO b SELECT * FROM x;
```
should not be accepted

Reply:
Added those scenarios, thanks!